microsimulation-rs/src/household.rs

```diff
@@ -0,0 +1,102 @@
+use serde::{Deserialize, Serialize};
+
+use crate::{person::PID, Eth, ID, OA};
+
+type UInt = u32;
+type Int = i32;
+
+#[derive(Debug, Serialize, Deserialize, Hash, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub struct HID(pub ID);
+impl std::fmt::Display for HID {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "HID #{}", self.0)
+    }
+}
+impl From<ID> for HID {
+    fn from(value: ID) -> Self {
+        Self(value)
+    }
+}
+
+pub fn serialize_bool<S>(bool: &Option<bool>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::ser::Serializer,
+{
+    let s = match bool {
+        Some(true) => "True",
+        Some(false) => "False",
+        None => "False",
+    };
+    serializer.serialize_str(s)
+}
+
+pub fn serialize_hrpid<S>(pid: &Option<PID>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::ser::Serializer,
+{
+    let s = match pid {
+        Some(pid) => format!("{}", pid.0),
+        None => "-1".to_string(),
+    };
+    serializer.serialize_str(&s)
+}
+
+// TODO: refactor field names to generic ones
```

Comment:
This would be useful for facilitating updates to new census data. Move to a new issue.